### PR TITLE
Update documentation to add ingress and egress filters

### PR DESCRIPTION
**Pull request by irenarindos**

### CODE REVIEW

File: website/content/docs/concepts/domain-model/targets.mdx [hunk 1]
```diff
@@ -58,13 +58,53 @@ TCP targets have the following additional attributes:
...
+  for this target.
+
+- `ingress_worker_filter` - (optional) <sup>HCP Only</sup>
+  A boolean expression to [filter][] which ingress workers can handle sessions
+  for this target.
+
+### SSH Target Attributes <sup>HCP Only</sup>
+
+SSH targets can source username/password or SSH private key credentials from Vault [credential libraries][] or static
+[credentials][]. Credentials are then injected into the SSH session between a client and end host. This allows users to
```
Comment (Dan-Heath) [inline website/content/docs/concepts/domain-model/targets.mdx:72]:
```suggestion
[credentials][]. Boundary then injects credentials into the SSH session between a client and end host. This allows users to
```

File: website/content/docs/concepts/domain-model/targets.mdx [hunk 2]
```diff
@@ -58,13 +58,53 @@ TCP targets have the following additional attributes:
...
+securely SSH to remote hosts while never being in possession of a valid credential for that target host.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/domain-model/targets.mdx:73]:
```suggestion
securely connect to remote hosts using SSH, while never being in possession of a valid credential for that target host.
```

File: website/content/docs/concepts/filtering/worker-tags.mdx [hunk 3]
```diff
@@ -137,8 +137,14 @@ Following are some examples of using these values in filters:
 # Target Worker Filtering
 
 Once workers have tags, these tags may be used to control which
-workers are allowed to manage a given session by specifying a `worker_filter`
-attribute when configuring [targets](/docs/concepts/domain-model/targets).
+workers are allowed to manage a given session by specifying optional worker filter attributes
+when configuring [targets](/docs/concepts/domain-model/targets).
+
+The `egress_worker_filter` attribute controls which workers will be used for egress to a target. This is the worker
+that will access the target.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/filtering/worker-tags.mdx:144]:
```suggestion
The `egress_worker_filter` attribute controls which workers are used for egress to a target. This is the worker
that accesses the target.
```

File: website/content/docs/concepts/filtering/worker-tags.mdx [hunk 4]
```diff
@@ -137,8 +137,14 @@ Following are some examples of using these values in filters:
...
+
+The `ingress_worker_filter`<sup>HCP Only</sup> attribute controls which workers will be used for ingress to a target.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/filtering/worker-tags.mdx:146]:
```suggestion
The `ingress_worker_filter`<sup>HCP Only</sup> attribute controls which workers are used for ingress to a target.
```

File: website/content/docs/concepts/filtering/worker-tags.mdx [hunk 5]
```diff
@@ -137,8 +137,14 @@ Following are some examples of using these values in filters:
...
+This is the worker a client will connect to when initiating a connection.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/filtering/worker-tags.mdx:147]:
```suggestion
This is the worker a client connects to when initiating a connection.
```

File: website/content/docs/concepts/domain-model/targets.mdx [hunk 6]
```diff
@@ -58,13 +58,53 @@ TCP targets have the following additional attributes:
...
+
+SSH targets have the following additional attributes:
+
+- `session_max_seconds` - (required)
+  The maximum duration of an individual session between the user and the target.
+  All connections for a session are closed
+  and the session is terminated
+  when a session reaches the maximum duration.
+  The default is 8 hours (28800 seconds).
+  Must be greater than 0.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/domain-model/targets.mdx:83]:
```suggestion
  This value must be greater than 0.
```

File: website/content/docs/concepts/domain-model/targets.mdx [hunk 7]
```diff
@@ -58,13 +58,53 @@ TCP targets have the following additional attributes:
...
+
+- `session_connection_limit` - (required)
+  The cumulative number of TCP connections allowed during a session.
+  -1 means no limit.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/domain-model/targets.mdx:87]:
```suggestion
 A -1 value means no limit.
```

File: website/content/docs/concepts/filtering/worker-tags.mdx [hunk 8]
```diff
@@ -137,8 +137,14 @@ Following are some examples of using these values in filters:
 # Target Worker Filtering
 
 Once workers have tags, these tags may be used to control which
```
Comment (Dan-Heath) [inline website/content/docs/concepts/filtering/worker-tags.mdx:139]:
```suggestion
Once workers have tags, you can use these tags to control which
```

File: website/content/docs/concepts/filtering/worker-tags.mdx [hunk 9]
```diff
@@ -137,8 +137,14 @@ Following are some examples of using these values in filters:
...
-workers are allowed to manage a given session by specifying a `worker_filter`
-attribute when configuring [targets](/docs/concepts/domain-model/targets).
+workers are allowed to manage a given session by specifying optional worker filter attributes
+when configuring [targets](/docs/concepts/domain-model/targets).
```
Comment (Dan-Heath) [inline website/content/docs/concepts/filtering/worker-tags.mdx:141]:
```suggestion
when you configure [targets](/docs/concepts/domain-model/targets).
```

File: website/content/docs/concepts/filtering/worker-tags.mdx (hunk 5 above)
Comment (Dan-Heath) [reply website/content/docs/concepts/filtering/worker-tags.mdx:147]: This sentence sounds a little funny to me with the repetition of the verb "to connect". I'm wondering if we can be more specific. Would it be accurate to say:

> This is the worker a client connects to when initiating a connection **to a target**.

**Review by Dan-Heath — COMMENTED** [review]
I added a few suggestions and a question, but overall I think this reads well and makes sense. Please let me know if you have any questions. Thank you for adding this to the docs!!

File: website/content/docs/concepts/domain-model/targets.mdx (hunk 1 above)
Comment (irenarindos) [reply website/content/docs/concepts/domain-model/targets.mdx:72]: thanks!

File: website/content/docs/concepts/domain-model/targets.mdx (hunk 2 above)
Comment (irenarindos) [reply website/content/docs/concepts/domain-model/targets.mdx:73]: thanks!!

File: website/content/docs/concepts/filtering/worker-tags.mdx (hunk 3 above)
Comment (irenarindos) [reply website/content/docs/concepts/filtering/worker-tags.mdx:144]: thanks!

File: website/content/docs/concepts/filtering/worker-tags.mdx (hunk 5 above)
Comment (irenarindos) [reply website/content/docs/concepts/filtering/worker-tags.mdx:147]: I like that! Thanks!

File: website/content/docs/concepts/domain-model/targets.mdx (hunk 7 above)
Comment (irenarindos) [reply website/content/docs/concepts/domain-model/targets.mdx:87]: thanks!

File: website/content/docs/concepts/domain-model/targets.mdx (hunk 6 above)
Comment (irenarindos) [reply website/content/docs/concepts/domain-model/targets.mdx:83]: thanks!

File: website/content/docs/concepts/filtering/worker-tags.mdx (hunk 8 above)
Comment (irenarindos) [reply website/content/docs/concepts/filtering/worker-tags.mdx:139]: thanks!

**Review by Dan-Heath — APPROVED** [review]
LGTM, thank you Irena!

**Comment by xingluw** [comment]
@irenarindos @Dan-Heath thanks for adding this, can we also include some information on default behavior? What will happen if one or more of the filters are not set

**Pull request merged**